tests/src/test/scala/com/nvidia/spark/rapids/DecimalBinaryOpSuite.scala

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import ai.rapids.cudf.DType
+
+import org.apache.spark.sql.rapids.{GpuAdd, GpuEqualTo, GpuGreaterThan, GpuGreaterThanOrEqual, GpuLessThan, GpuLessThanOrEqual, GpuMultiply, GpuSubtract}
+import org.apache.spark.sql.types.{DataType, DataTypes, Decimal, DecimalType, StructType}
+
+/*
+  unsupported operators are as below:
+  - GpuEqualNullSafe
+  - GpuDivModLike (GpuDivide/GpuIntegralDivide/GpuPmod/GpuRemainder)
+*/
+class DecimalBinaryOpSuite extends GpuExpressionTestSuite {
```

Comment:
I personally would prefer to see us update the python tests.  They tend to be more complete.  These tests are not covering the case where you need to take a literal and turn it into a column because https://github.com/rapidsai/cudf/pull/6723 has not been merged in yet.

This is another reason to break down this into smaller pieces.  I cannot let us put in supporting for binary ops without the above PR being merged in and tested.  If we break this apart we can get some of the functionality in, and then we can look at what we need to do to get the rest of it in.